:sectnumlevels: attribute documented

jaredmorgs opened the pull request:
Resolves Issue #383 by adding the :sectnumlevels: attribute to the documentation. Based the command line and API guidance on the :sectnums: attribute, I added similar override information as this seemed to be important in #288.

mojavelinux commented:
> Based the command line and API guidance on the :sectnums: attribute, I added similar override information...

Actually, this is not true. Toggling this value midstream is not recognized. `sectnumlevels` is a "header only" attribute, which means it has no meaning outside of the document header.

(Btw, the only attribute that doesn't get "locked" when set via the API or CLI is `sectnums`. It's unique in this way. We call this a "flexible attribute" because it can always be changed midstream no matter how it is declared).

mojavelinux commented on docs/_includes/sections.adoc:
```diff
@@ -170,13 +171,18 @@ If `sectnums` is set on the command line (or API), that overrides the value set
+
+If `sectnumlevels` is set on the command line (or API), it overrides the value set in the document header, but it does not prevent the document from toggling the value for regions of the document.
+
+If `sectnumlevels!` is set on the command line (or API), the numbers are disabled regardless of toggling within the document.
```
These points about toggling the attribute value doesn't apply for `sectnumlevels`.

jaredmorgs commented:
Ahh, good to know. I'll correct the info and push another PR. Are you happy
with the rest of the content?

On Mon, 18 Jan 2016 at 07:52 Dan Allen <notifications@github.com> wrote:

> Based the command line and API guidance on the :sectnums: attribute, I
> added similar override information...
>
> Actually, this is not true. Toggling this value midstream is not
> recognized. sectnumlevels is a "header only" attribute, which means it
> has no meaning outside of the document header.
>
> (Btw, the only attribute that doesn't get "locked" when set via the API or
> CLI is sectnums. It's unique in this way. We call this a "flexible
> attribute" because it can always be changed midstream no matter how it is
> declared).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asciidoctor/asciidoctor.org/pull/477#issuecomment-172386534>
> .
>
-- 

Sent from Mobile.

mojavelinux commented on docs/_includes/sections.adoc:
```diff
@@ -170,13 +171,18 @@ If `sectnums` is set on the command line (or API), that overrides the value set
 
-////
-You can also restrict section numbering with the +sectnumlevels+ attribute.
+=== Numbering Depth
```
We should use sentence capitalization for titles (though, admittedly, it's not consistent atm).

mojavelinux commented on docs/_includes/sum-section.adoc:
```diff
@@ -34,4 +34,10 @@ Included in:
 Can be toggled on or off through document.
+
+|sectnumlevels
+|1-9
```
The correct range is 0-5. (0 is effectively equivalent to disabling section numbering).

mojavelinux commented:
I just finished the docs review.

jaredmorgs commented:
Thanks for the feedback, Dan. Will make these changes and update the
original commit. :D

On Mon, 18 Jan 2016 at 07:58 Dan Allen <notifications@github.com> wrote:

> I just finished the docs review.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asciidoctor/asciidoctor.org/pull/477#issuecomment-172387515>
> .
>
-- 

Sent from Mobile.

mojavelinux commented:
Super! Let me know when it's ready and I'll do another pass at it. :memo:

jaredmorgs commented:
Git is great when it works as you expected, but when it doesn't with regard to git --amend, you get the chook's breakfast you see above.  

In summary:

Resolves Issue #383 by further adding to the existing sectnumlevels attribute docs. Removed override info, added detail about Flexible Attribute to sectnums (it's worth noting), and declared the `setnumlevels` attribute as 'intrinsic' because based on the current guidance about attributes in  typeshttp://asciidoctor.org/docs/user-manual/#attributes, it appears as though `sectnumlevels` falls into this category.

mojavelinux commented:
Nice work!

> Git is great when it works as you expect, but when it doesn't with regard to git --amend

Revising commits is something that is tricky to learn, but once you do, you'll realize that just a few commands give you everything you need. The key thing to remember is that git is just about passing around changes. Every commit is just a delta. Using git --amend or rebase, you are just telling it how to organize those deltas into a different batch. If everything goes absolutely horribly wrong, you can just diff two versions to get the effective change, store that somewhere, reset the branch and apply the changes.

Just always remember that git is just sets of changes. And those changes can be redefined. At least, once I saw it that way, it wasn't a black box for me anymore.

I'll do a rebase merge on this one to clean the history a bit.

mojavelinux commented:
@jaredmorgs take note of a few conventions I followed in the revised commit https://github.com/asciidoctor/asciidoctor.org/commit/4b054755bb683fc17075a72aa900c1827c85d48c

* how I refer to the issue in the commit subject (first line) (`resolves #xyz subject line`)
* how I used an unordered list (Markdown syntax) to summarize the changes

I find this works really well and renders well on GitHub. That's the style we want to document in the contribution guide and consistently follow.

jaredmorgs commented:
Noting this for the contributors guide. 👍

On Tue, 19 Jan 2016, 16:48 Dan Allen <notifications@github.com> wrote:

> Closed #477 <https://github.com/asciidoctor/asciidoctor.org/pull/477>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asciidoctor/asciidoctor.org/pull/477#event-518720464>.
>
-- 

Sent from Mobile.
